logserver/src/main/java/alluxio/logserver/AlluxioLogServerProcess.java

```diff
@@ -0,0 +1,142 @@
+ * be stored.
+ */
+public class AlluxioLogServerProcess implements LogServerProcess {
+  private final static Logger LOG = LoggerFactory.getLogger(AlluxioLogServer.class);
+  private ServerSocket mServerSocket;
+  private int mPort;
+  private String mBaseLogDir;
+  private Map<InetAddress, Hierarchy> mInetAddressHashMap;
+  private boolean mStopped;
+
+  public AlluxioLogServerProcess(String portStr, String baseLogDir) {
+    try {
+      mPort = Integer.parseInt(portStr);
+      mBaseLogDir = baseLogDir;
+      mServerSocket = new ServerSocket(mPort);
+      mInetAddressHashMap = new HashMap<>();
+      mStopped = false;
+    } catch (Exception e) {
+      throw new RuntimeException(e);
+    }
+  }
+  /**
+   * Starts the Alluxio process. This call blocks until the process is stopped via
+   * {@link #stop()}. The {@link #waitForReady()} method can be used to make sure that the
+   * process is ready to serve requests.
+   */
+  @Override
+  public void start() throws Exception {
+    LOG.info("Log server started.");
+    while (!mStopped) {
+      Socket client = mServerSocket.accept();
+      InetAddress inetAddress = client.getInetAddress();
+      // Currently loggerRepository is always null because we are not using the map.
+      LoggerRepository loggerRepository = mInetAddressHashMap.get(inetAddress);
+      if (loggerRepository == null) {
+        loggerRepository = configureHierarchy(inetAddress);
+      }
+      try {
+        new Thread(new AlluxioLog4jSocketNode(client, loggerRepository)).start();
+        LOG.info("Client: {} connected.", inetAddress.toString());
+      } catch (IOException e) {
+        // Could not create a thread to serve a client, ignore this client.
+        LOG.warn("Failed to connect with client: {}.", inetAddress.toString());
+        continue;
+      }
+    }
+  }
+
+  /**
+   * Stops the Alluxio process, blocking until the action is completed.
+   */
+  @Override
+  public void stop() throws Exception {
+    mStopped = true;
+    LOG.info("Log server stopped.");
+  }
+
+  /**
+   * Waits until the process is ready to serve requests.
+   */
+  @Override
+  public void waitForReady() {
+    CommonUtils.waitFor(this + " to start", new Function<Void, Boolean>() {
+      @Override
+      public Boolean apply(Void input) {
+        return mServerSocket != null;
+      }
+    }, WaitForOptions.defaults().setTimeoutMs(10000));
+  }
+
+  private LoggerRepository configureHierarchy(InetAddress inetAddress) throws IOException, URISyntaxException {
+    Hierarchy clientHierarchy;
+    String inetAddressStr = inetAddress.toString();
+    int i = inetAddressStr.indexOf("/");
+    String key;
+    if (i == 0) {
+      key = inetAddressStr.substring(1, inetAddressStr.length());
+    } else {
+      key = inetAddressStr.substring(0, i);
+    }
+    Properties properties = new Properties();
+    File configFile = new File(new URI(System.getProperty("log4j.configuration")));
```

Comment:
Make this a constant?

Reply:
Do you mean a static final of the class?